sys/include/net/nanocoap.h

```diff
@@ -322,6 +322,33 @@ typedef const struct {
     const size_t resources_numof;       /**< number of entries in array */
 } coap_resource_subtree_t;
 
+/**
+ * @brief   Size of the CoAP request context struct
+ */
+#define COAP_REQUEST_CTX_SIZE   (2 * sizeof(void *) + \
+                                 IS_USED(MODULE_GCOAP) * sizeof(uint32_t))
+
+/**
+ * @brief   Define and initialize CoAP request context struct
+ *
+ * @param[in] ctx    Name of the request context variable
+ * @param[in] remote Remote endpoint that made the request
+ */
+#define COAP_REQUEST_CTX_INIT(ctx, remote)                      \
```

Comment:
sprint discussion: @kaspar030 strongly advises against macros. this already breaks alignment.
idea: move nanocoap_internal.h back into nanocoap.h, no macro needed anymore.

Reply:
Ok, I dropped `nanocoap_internal.h`.